tests/functional/u/use/used_before_assignment_issue85.py

```diff
@@ -7,3 +7,61 @@ def main():
     finally:
         print(res)  # [used-before-assignment]
     print(res)
+
+
+def try_except_finally():
+    """When evaluating finally blocks, assume try statements fail."""
+    try:
+        res = 1 / 0
+        res = 42
+    except ZeroDivisionError:
+        print()
+    finally:
+        print(res)  # [used-before-assignment]
+    print(res)
+
+
+def try_except_finally_assignment_in_final_block():
+    """Assignment of the name in the final block does not warn."""
+    try:
+        res = 1 / 0
+        res = 42
+    except ZeroDivisionError:
+        print()
+    finally:
+        res = 999
+        print(res)
+    print(res)
+
+
+def try_except_finally_nested_try_finally_in_try():
+    """Don't confuse assignments in different finally statements where
+    one is nested inside a try.
+    """
+    try:
+        try:
+            res = 1 / 0
+        finally:
+            print(res)  # [used-before-assignment]
+        print(1 / 0)
+    except ZeroDivisionError:
+        print()
+    finally:
+        res = 999  # this assignment could be confused for that above
+        print(res)
+    print(res)
+
+
+def try_except_finally_nested_try_finally_in_finally():
+    """Don't confuse assignments in different finally statements where
+    one is nested inside a finally.
+    """
+    try:
+        pass
+    finally:
+        try:
+            times = 1
+        except TypeError:
+            pass
+        # Don't emit: only assume the outer try failed
+        print(times)
```

Comment:
I don't understand ``Don't confuse assignments in different finally statements``. Shouldn't we raise or ``times`` be defined somewhere in the other finally statement?

Reply:
The docstring looks like it was lazily copied from above and not really updated. I'll push again and explain in the docstring how I see what is being tested and where to go from here.

Reply:
In rewriting the tests I discovered I could handle nesting better. So there are more tests now :-)